rocAL_pybind/amd/rocal/plugin/pytorch.py

```diff
@@ -270,6 +270,111 @@ def __init__(self,
+    With the data set [1,2,3,4,5,6,7] and the batch size 2:
+    fill_last_batch = False, last_batch_padded = True  -> last batch = [7], next iteration will return [1, 2]
+    fill_last_batch = False, last_batch_padded = False -> last batch = [7], next iteration will return [2, 3]
+    fill_last_batch = True, last_batch_padded = True   -> last batch = [7, 7], next iteration will return [1, 2]
+    fill_last_batch = True, last_batch_padded = False  -> last batch = [7, 1], next iteration will return [2, 3]
+
+    """
+    def __init__(self, pipeline, tensor_layout = types.NCHW, reverse_channels = False, multiplier = [1.0, 1.0, 1.0], offset = [0.0, 0.0, 0.0], tensor_dtype = types.FLOAT, size = -1, auto_reset = False, device = "cpu", device_id = 0):
+        self.loader = pipeline
+        self.tensor_format = tensor_layout
+        self.multiplier = multiplier
+        self.offset = offset
+        self.reverse_channels = reverse_channels
+        self.device = device
+        self.device_id = device_id
+        self.output = None
+        self.iterator_length = b.getRemainingImages(self.loader._handle)
+        self.max_shape = None
+        self.batch_size = self.loader._batch_size
+        self.output_list = None
+        self.labels_size = self.batch_size
+        self.output_memory_type = self.loader._output_memory_type
+
+
+    def next(self):
+        return self.__next__()
+
+    def __next__(self):
+        if self.loader.rocal_run() != 0:
+            raise StopIteration
+        else:
+            self.output_tensor_list = self.loader.get_output_tensors()
+
+        self.output_list = []
+        for i in range(len(self.output_tensor_list)):
+            dimensions = self.output_tensor_list[i].dimensions()
+            self.num_roi_dims = len(dimensions) - 1
+            self.roi_array = np.zeros(self.batch_size * self.num_roi_dims * 2, dtype=np.int32)
+            self.output_tensor_list[i].copy_roi(self.roi_array)
+            if self.device == "cpu":
+                torch_dtype = self.output_tensor_list[i].dtype()
+                output = torch.empty(dimensions, dtype=getattr(torch, torch_dtype))
+                self.labels_tensor = torch.empty(self.labels_size, dtype=getattr(torch, torch_dtype))
+            else:
+                torch_gpu_device = torch.device('cuda', self.device_id)
+                torch_dtype = self.output_tensor_list[i].dtype()
+                output = torch.empty(dimensions, dtype=getattr(torch, torch_dtype), device=torch_gpu_device)
+                self.labels_tensor = torch.empty(self.labels_size, dtype=getattr(torch, torch_dtype), device=torch_gpu_device)
+
+            self.output_tensor_list[i].copy_data(ctypes.c_void_p(output.data_ptr()), self.output_memory_type)
+            self.output_list.append(output)
+
+
+        # self.labels = self.loader.get_image_labels() #Uncomment when meta-data is added
+        # self.labels_tensor = self.labels_tensor.copy_(torch.from_numpy(self.labels)).long()
```

Comment:
Are these commented lines needed?

Reply:
These lines needs to be uncommented when meta-data PR is introduced. I will remove it from this PR as of now